control/mateqn.py

```diff
@@ -220,17 +221,17 @@ def lyap(A, Q, C=None, E=None):
         if (size(Q) > 1 and shape(Q)[0] != shape(Q)[1]) or \
            (size(Q) > 1 and shape(Q)[0] != n) or \
            (size(Q) == 1 and n > 1):
-            raise ControlArgument("Q must be a square matrix with the same \
+            raise ControlDimension("Q must be a square matrix with the same \
                 dimension as A.")
 
         if (size(E) > 1 and shape(E)[0] != shape(E)[1]) or \
            (size(E) > 1 and shape(E)[0] != n) or \
            (size(E) == 1 and n > 1):
-            raise ControlArgument("E must be a square matrix with the same \
+            raise ControlDimension("E must be a square matrix with the same \
                 dimension as A.")
 
         if not _is_symmetric(Q):
-            raise ControlArgument("Q must be a symmetric matrix.")
+            raise ControlDimension("Q must be a symmetric matrix.")
```

Comment:
That's not a dimension error. `ControlDimension` would have been raised above.

Reply:
True. I was trying to make it so both slycot and scipy versions raised the same kind of error (ValueError) rather than different types (ControlArgument is a TypeError), so that we could run the same test on both. win-win solution: I'll just change that to a ValueError because I think it can be argued that being non-symmetric is more of a value error than a type error.